openlibrary/accounts/__init__.py

```diff
@@ -11,6 +11,32 @@ def get_group(name):
     return web.ctx.site.get("/usergroup/%s"%name)
 
 
+def escalate_privilege_and_run_as(username, action, kwargs=None):
```

Comment:
Instead of a lambda function, this would be a great usecase for python's `with` statements:

e.g.
```py
with accounts.run_as('/user/foo'):
    do_stuff()
```